datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -794,13 +794,20 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 op: Divide,
                 right,
             } if is_null(&right) => *right,
-            // A / A --> 1 (if a is not nullable)
+            // 0 / 0 -> null
             BinaryExpr {
                 left,
                 op: Divide,
                 right,
-            } if !info.nullable(&left)? && left == right => lit(1),
-
+            } if is_zero(&left) && is_zero(&right) => Expr::Literal(ScalarValue::Int32(None)),
+            // A / 0 --> DivideByZeroError 
+            BinaryExpr {
+                left: _,
+                op: Divide,
+                right,
+            } if is_zero(&right) => {
```

Comment:
Should this be `!info.nullable(&left)? && is_zero(&right)` like the modulo case?

Reply:
You are right! I should have checked if it is nullable or not just like the modulo case. Thanks.

Reply:
Thanks. I will add `!info.nullable(&left)? && is_zero(&right)` check.